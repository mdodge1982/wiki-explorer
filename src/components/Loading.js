import React, {Component} from 'react';

class Loading extends Component {
	constructor(props) {
		super(props);
		const int = setInterval(() => {
			let newContent = '';
			if(this.state.content!=='...'){
				newContent = this.state.content+'.';
			}
			this.setState({content: newContent});
		},100);
		this.state = {
			content: '',
			int
		};
	}
	render() {
		return (
			<span>{this.state.content}</span>
		)
	}
	componentWillUnmount() {
		console.log('component unmount');
		clearInterval(this.state.int);
	}
}

export default Loading;
