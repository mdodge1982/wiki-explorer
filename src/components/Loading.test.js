import React from 'react';
import Loading from './Loading';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';

test('Loader starts empty', () => {
	const component = renderer.create(<Loading/>);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();

});

test('Loader adds dots', () => {

	jest.useFakeTimers();

	const component = ReactTestUtils.renderIntoDocument(<Loading/>)
	jest.runTimersToTime(100);
	expect(component.state.content).toEqual('.');
	jest.runTimersToTime(100);
	expect(component.state.content).toEqual('..');
	jest.runTimersToTime(100);
	expect(component.state.content).toEqual('...');

});
