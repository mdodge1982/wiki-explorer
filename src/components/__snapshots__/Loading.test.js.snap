// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader starts empty 1`] = `
<span>
  
</span>
`;
